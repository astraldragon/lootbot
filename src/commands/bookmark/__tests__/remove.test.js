jest.mock("discord-akairo");
jest.mock("../../../logger");
jest.mock("../../../models", () => ({
  Bookmark: { destroy: jest.fn("destroy") }
}));

const logger = require("../../../logger");
const { Bookmark } = require("../../../models");
const { Command } = require("discord-akairo");
const { checkManagePermissions } = require("../../../support");
const RemoveCommand = require("../remove");

describe("commands/bookmark/remove", () => {
  beforeEach(() => {
    this.command = new RemoveCommand();

    this.args = {
      role: { id: "1", name: "Test" }
    };

    this.msg = {
      guild: { id: "1234" },
      channel: {
        send: jest.fn("send").mockReturnValue(Promise.resolve())
      }
    };
  });

  it("should configure properly", () => {
    expect(Command).toMatchSnapshot();
  });

  it("should have the manage permission check", () => {
    expect(Command.mock.calls[0][1].options.permissions).toBe(
      checkManagePermissions
    );
  });

  it("should destroy the passed in role/bookmark", async () => {
    Bookmark.destroy.mockReturnValue(1);
    await this.command.exec(this.msg, this.args);

    expect(Bookmark.destroy).toHaveBeenCalledWith({
      where: { role: this.args.role.id, guild: this.msg.guild.id }
    });

    expect(this.msg.channel.send).toHaveBeenCalledWith(
      `${this.args.role.name} tracking stopped`
    );
  });

  it("should notify when the passed in role/bookmark does not exist", async () => {
    Bookmark.destroy.mockReturnValue(0);
    await this.command.exec(this.msg, this.args);

    expect(this.msg.channel.send).toHaveBeenCalledWith(
      `${this.args.role.name} not found`
    );
  });

  it("should notify if any errors occur", async () => {
    Bookmark.destroy.mockImplementation(() => {
      throw new Error("test");
    });

    await this.command.exec(this.msg, this.args);

    expect(logger.error).toHaveBeenCalledWith("test");
    expect(this.msg.channel.send).toHaveBeenCalledWith(
      `An error occurred removing ${this.args.role.name}`
    );
  });
});
