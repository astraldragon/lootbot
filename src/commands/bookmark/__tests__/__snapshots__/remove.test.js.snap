// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commands/bookmark/remove should configure properly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "bookmark-remove",
      Object {
        "aliases": Array [
          "bookmark-remove",
          "br",
        ],
        "args": Array [
          Object {
            "id": "role",
            "prompt": Object {
              "start": "What is the role to stop tracking?",
            },
            "type": "role",
          },
        ],
        "category": "Bookmark",
        "channelRestriction": "guild",
        "description": Object {
          "content": "Remove bookmark tracking from a role",
          "examples": Array [
            "bookmark-remove \\"Some Role\\"",
          ],
          "usage": "<role>",
        },
        "options": Object {
          "permissions": [Function],
        },
        "split": "quoted",
      },
    ],
  ],
}
`;
